src/dynamics/clock.jl

```diff
@@ -38,6 +38,18 @@ function Base.show(io::IO, C::Clock)
     print_fields(io, C, keys)
 end
 
+# copy! 
+function Base.copy!(clock::Clock, clock_old::Clock)
+    clock.time = clock_old.time 
+    clock.start = clock_old.start 
+    clock.period = clock_old.period 
+    clock.timestep_counter = clock_old.timestep_counter
+    clock.n_timesteps = clock_old.n_timesteps
+    clock.Δt = clock_old.Δt 
+    
+    return nothing 
+end 
```

Comment:
Looks good, but why is that needed? I'm suspecting to restart a simulation? Cause in that case we'd need to check whether all of them are supposed to just be copied over or whether some of them should be set differently afterwards

Reply:
As you see in the code of the test, we really need to copy the structs quite often. I also sometimes use out-of-place forms of the functions, so my intend was to really have them identical if I copy them. Intuitively that's also what I would expect from a `copy!` function. 
I am happy to revert the change though, I also realise now that it's a bit more directly related to the integration. But maybe we want to have it like this? Because before a new integration is started `set_period` and `initialise!(::Clock)` is called again anyway. So this doesn't break anything as far as I can see it.

Reply:
No I agree that's what I expect a `copy!` function to do, but you know that `deepcopy` is always defined for all types?
```julia
julia> struct S
           a
       end

julia> s = S(1)
S(1)

julia> deepcopy(s)
S(1)
```

Reply:
yeah,`deepcopy` is very generic and works for all types, `copy` has to be specifically defined.